homeassistant/components/synology_dsm/common.py

```diff
@@ -94,6 +94,13 @@ async def async_setup(self) -> None:
             self._with_surveillance_station,
         )
 
+        # check if upgrade is available
+        try:
+            await self._hass.async_add_executor_job(self.dsm.upgrade.update)
```

Comment:
There are 3 executor jobs in this function.  Can we combine them into a single call?

Reply:
unfortunately, i do not think so 🙁  the `_fetch_device_configuration()`  could be done async, so one executer job will be moved into another function 🙈

Reply:
as long as the library is not migrated to aiohttp, we still need to rely on executer jobs here

Reply:
ahhh ... you want to pack all three calls into an own function and than call this once in an executer?

Reply:
many thanks @bdraco for discussion about this and showing me this great improvement 👍